project/Dependencies.scala

```diff
@@ -53,7 +53,8 @@ object Dependencies {
   lazy val test = Seq(
     "junit" % "junit" % "4.13.2" % "test",
     "com.github.sbt" % "junit-interface" % "0.13.3" % "test",
-    "org.scalacheck" %% "scalacheck" % "1.18.1" % "test"
+    "org.scalacheck" %% "scalacheck" % "1.18.1" % "test",
+    "com.lihaoyi" %% "sourcecode" % "0.4.1"
```

Comment:
To clarify, to us the tdml-lib library is "main" code. Our users will incorporate that dependency as a "test" dependency.

Reply:
This should not be added to 'val test'. `test` dependencies are added to all jars. This wants to only be a dependency of daffodil-tdml. We probably want a special `val tdml = Seq(...sourcecode)`, which is added to the project.

Reply:
Yeah, I think we need another definition in Dependencies.scala for use with just tdml specific dependencies, and it will have just this one.

Reply:
@stevedlawrence  This is just a dependency for the tdml-lib now. 

Li Haoyi's library is MIT license, which is Apache compatible. 

While the CLI does run TDML tests, and so depends on daffodil-tdml-lib, it isn't dependent on this feature, which is only used by Scala code that explicitly creates calls to do the Runner.doTest method, which depends on this sourcecode macro which is passed through to scala code users in the implicit argument to doTest. 

But should we add info about this library anyway just to be safe? I am not sure CLI won't pull it in. I imagine it will just depend on everything that daffodil-tdml-lib depends on? 

This would mean adding to the CLI's bin.LICENSE file I believe.

Is there anyplace else to update on account of this dependency?

Reply:
If `sourcecode` is a transitive dependency of the CLI via daffdodil-tdml, then we *must* include license/notice information in `daffodil-cli/bin.{LICENSE,NOTICE)`. All CLI transitive dependencies are packaged/distributed in our CLI helper binaries whether they are used or not, and anything packaged/distributed in there needs to be mentioned in the the CLI specific license files. Transitive dependencies do not need to be listed anywhere else since we don't distribute them, as long they are ASF compatible or optional they are fine.

Note: See my other thought about whether this really wants to be part of daffodil-sbt. It's pretty easy to add new dependencies via daffodil-sbt.

Reply:
The more I think about this, I don't think this wants to be a source dependency. It really just wants to be a "test" dependency for use by JUnit files. If we want to also use it for schema projects, then we should modify daffodil-sbt to also add it as a "test" dependency. That avoids the awkwardness of distributing something with the CLI that is never used by the CLI. There's a bit of duplication, but it makes it so we can use the `sourcecode` approach (or the alternative junit approach) with any version of Daffodil for schema projects. Otherwise to test with an older `daffodilVersion` you would need to revert `runner.doTest` since `sourcecode` would only be pulled in for newer versions of daffodil-lib. The fix is to then manually add `sourcecode` dependency for those projects, but that's a pain when we could just add support in daffodil-sbt.

Reply:
Yeah, since the daffodil-sbt plugin takes care of the dependency on daffodil-tdml-processor, it has to add the "test" qualification now.  

A daffodil-tdml-junit library is sounding more appealing even without daffodil-sbt support for it. 

In fact our daffodil libraries really do need to be usable without daffodil-sbt. E.g., suppose someone's corporate standard is maven or other tool. We need all the daffodil libraries including the tdml runner, etc. all to be usable without depending on daffodil-sbt.  Having a isolated library where all the junit dependencies for gluing together tdml and junit testing makes sense.